torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1146,7 +1146,7 @@ class ShapePropagator {
     //   - has a bool keepdim argument
     static const register_formula_for dim_reduce_ops_with_integer_upcast{
```

Comment:
In these `register_formula_for` entries it looks like the code is explicitly parsing each argument in the registered function. It seems that since I've added a dtype argument, I also need to parse that. I can guess at what the code should look like to do so, but I don't have a failing tests and I'm not sure how to write one.